sql/plan/trigger.go

```diff
@@ -190,6 +190,25 @@ func shouldUseLogicResult(logic sql.Node, row sql.Row) (bool, sql.Row) {
 			})
 		}
 		return hasSetField, row[len(row)/2:]
+	case *BeginEndBlock:
```

Comment:
You probably already figured this out but this entire technique is super sketchy and probably doesn't work for a bunch of things. You'll probably need to rewrite it for actual stored procedures.

Reply:
Yeah I thought of some other cases that I may need to handle after I submitted the PR, but I'll get to those.